torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3096,10 +3097,15 @@ struct to_ir {
     // Set output type from FunctionSchema and corresponding rpc_op.
     const std::vector<Argument>& returns = functionSchema.returns();
     TORCH_INTERNAL_ASSERT(returns.size() == 1);
-    auto output_type = returns[0].type();
+    TypePtr output_type = nullptr;
     if (rpc_op == prim::rpc_async) {
-      rpc_node_output->setType(FutureType::create(output_type));
+      output_type = FutureType::create(returns[0].type());
+    } else if (rpc_op == prim::rpc_remote) {
+      output_type = RRefType::create(returns[0].type());
+    } else {
+      output_type = returns[0].type();
     }
+    rpc_node_output->setType(output_type);
```

Comment:
This looks a little different from the previous version. Prior to this change, if `rpc_op == prim::rpc_sync`, `rpc_node_output->setType` is not called. Is it true that, by default, its type is already `returns[0].type()`?

Reply:
good catch, no it's not automatically set I think, let me change it in previous PR